### PR TITLE
isDiagonalPositive now acts as the documentation describes

### PR DESCRIPTION
**Pull request by lessthanoptimal**
isDiagonalNotNegative has been added with the same behavior as the original implementation

Thank you for your contribution to the EJML project.

Before submitting this PR, please make sure:
- [x] Your code is covered by tests
- [x] You ran `./gradlew spotlessJavaApply`

### CODE REVIEW

File: main/ejml-ddense/src/org/ejml/dense/row/CovarianceOps_DDRM.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public static boolean isValidFast( DMatrixRMaj cov ) {
      * @return 0 = is valid 1 = failed positive diagonal, 2 = failed on symmetry, 2 = failed on positive definite
```
Comment (TobiasLangner-TomTom) [inline main/ejml-ddense/src/org/ejml/dense/row/CovarianceOps_DDRM.java]: 1 = failed non-negative diagonal

Comment (TobiasLangner-TomTom) [reply main/ejml-ddense/src/org/ejml/dense/row/CovarianceOps_DDRM.java]: and 3 = failed on positive definite

Comment (lessthanoptimal) [reply main/ejml-ddense/src/org/ejml/dense/row/CovarianceOps_DDRM.java]: Thanks! Fixed and improved documentation.

File: main/ejml-ddense/src/org/ejml/dense/row/CovarianceOps_DDRM.java [hunk 2]
```diff
@@ -44,17 +44,18 @@ private CovarianceOps_DDRM(){}
      * @return true if valid and false if invalid
      */
     public static boolean isValidFast( DMatrixRMaj cov ) {
-        return MatrixFeatures_DDRM.isDiagonalPositive(cov);
+        return MatrixFeatures_DDRM.isDiagonalNotNegative(cov);
```
Comment (TobiasLangner-TomTom) [inline main/ejml-ddense/src/org/ejml/dense/row/CovarianceOps_DDRM.java:47]: javadoc: "if the diagonal elements are all not negative"

**Comment by lessthanoptimal** [comment]
@TobiasLangner-TomTom if there are still documentation errors we can fix it in the next update. Need to unblock the next release.

**Pull request merged**